### PR TITLE
Add --actor-config to snactor run

### LINKED ISSUE #530: Workflow configuration: missing snactor support
Current implementation of the workflow configuration is problematic when used with snactor. See https://github.com/oamg/leapp/pull/505 When we run e.g. `# snactor run <ActorName>` on actor which depends on workflow configuration model it fails. We need to find a way to provide the config to the actor upon issuing the snactor command. Additionally the exception/error message should be modified to reflect the change.

### PR DESCRIPTION
**Pull request by vinzenz**
This patch introduces support for specifying a model which the actor
should consume for configuration data.
Using this feature requires to run an actor before which produces this
model and saves the output.

Jira-Task: OAMG-1092

Signed-off-by: Vinzenz Feenstra <vfeenstr@redhat.com>
Fixes #530

### CODE REVIEW

File: leapp/snactor/commands/run.py [hunk 1]
```diff
@@ -19,10 +19,18 @@
 '''
 
 
+def _get_model(name, repo):
+    try:
+        return {model.__name__.lower(): model}[name.lower()]
```
Comment (Rezney) [inline leapp/snactor/commands/run.py:24]:
```
Traceback (most recent call last):
  File "/usr/bin/snactor", line 9, in <module>
    load_entry_point('leapp==0.10.0', 'console_scripts', 'snactor')()
  File "/usr/lib/python2.7/site-packages/leapp/snactor/__init__.py", line 78, in main
    cli.command.execute(version=_('snactor version {}').format(VERSION))
  File "/usr/lib/python2.7/site-packages/leapp/utils/clicmd.py", line 103, in execute
    args.func(args)
  File "/usr/lib/python2.7/site-packages/leapp/utils/clicmd.py", line 125, in called
    self.target(args)
  File "/usr/lib/python2.7/site-packages/leapp/utils/repository.py", line 20, in checker
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/leapp/snactor/context.py", line 32, in wrapper
    return f(*args, **kwargs)
  File "/usr/lib/python2.7/site-packages/leapp/snactor/commands/run.py", line 46, in cli
    config_model = _get_model(args.actor_config, repository) if args.actor_config else None
  File "/usr/lib/python2.7/site-packages/leapp/snactor/commands/run.py", line 24, in _get_model
    return {model.__name__.lower(): model}[name.lower()]
NameError: global name 'model' is not defined
```

**Comment by Rezney** [comment]
@vinzenz is this still WIP?

**Comment by Rezney** [comment]
Tested, seems to be working fine. @pirat89 any potential comments here?

**Comment by Rezney** [comment]
@vinzenz I originally thought the failed tests was due to some outage but:

~~~
=================================== FAILURES ===================================

_____________ [pylint] leapp/snactor/commands/messages/__init__.py _____________

W: 15, 0: Reimport 'requires_repository' (imported line 11) (reimported)

============= 1 failed, 98 passed, 179 deselected in 27.71 seconds =============
~~~

could you please take a look?

**Comment by drehak** [comment]
Is the option only meant to be used for `IPUConfig` or other models too?
What about using the option multiple times with different models? The corresponding `BaseMessaging` attribute, `_config_models` is a tuple, but if I use the option multiple times, only the latter one gets used. (In the case it's not meant to be used multiple times, shouldn't such usage throw an error instead of some of them being silently discarded?)

**Comment by Rezney** [comment]
@drehak It is meant just for "IPUConfig" model as this is used differently then other models. Please note that you still have to run "IPUWorkflowConfig" actor to generate the data, same as to get the messages from other actors, which however do not require any special treatment. HTH.

**Comment by drehak** [comment]
@Rezney if it's for and only for IPUConfig, shouldn't the option just be a flag? Being able to specify any model as an argument is kinda pointless, if the argument is going to be IPUConfig anyway.

**Comment by Rezney** [comment]
@drehak To give a real example. We may (or may not) want to migrate Convert2RHEL to LEAPP. In this case, there would be a new repository, with a new workflow, and a new config model C2RConfig. HTH

**Comment by drehak** [comment]
@Rezney Alright, looks good to me then.

**Comment by Rezney** [comment]
@vinzenz could you please also document this?

**Pull request merged**